### ISSUE TITLE
Using `RelativePanel.AlignTopWith` or `AlignVerticalCenterWith` inside a VisualState setter causes a WASM UI to not load

### ISSUE THREAD

**Issue by mrlacey** [unoplatform/uno#6124]
## Current behavior

If I include a VisualState setter that changes `RelativePanel.AlignTopWith` or `RelativePanel.AlignVerticalCenterWith` in the xaml of a page, that page will not be rendered in the WASM version of the app.
The app seems to start up and then shows nothing or (possibly) gets stuck in an infinite loop when trying to determine the correct layout.

That there's nothing in any debug or compilation output to indicate what the problem is makes it particularly hard to find the cause of these issues.

## Expected behavior

1. The RelativePanel should behave the same way on WASM as in UWP.

2. There should be no errors that produce no usable output or debug messages.

## How to reproduce it (as minimally and precisely as possible)

1. Create a new, blank app.
2. Replace the `Grid` in MainPage.xaml with the follwoing.

```
<RelativePanel>
    <VisualStateManager.VisualStateGroups>
        <VisualStateGroup>
            <VisualState>
                <VisualState.StateTriggers>
                    <AdaptiveTrigger MinWindowWidth="800" />
                </VisualState.StateTriggers>
                <VisualState.Setters>
                    <Setter Target="Secondary.(RelativePanel.RightOf)" Value="Primary" />
                    <!--If either of the following lines is included, the page won't load in a WASM app. comment it out and it will. --> 
                    <!--<Setter Target="Secondary.(RelativePanel.AlignTopWith)" Value="Primary" />-->
                    <!--<Setter Target="Secondary.(RelativePanel.AlignVerticalCenterWith)" Value="Primary"/>-->
                    </VisualState.Setters>
            </VisualState>
        </VisualStateGroup>
    </VisualStateManager.VisualStateGroups>
    <Rectangle x:Name="Primary" Width="400" Height="400" Fill="Red"/>
    <Rectangle x:Name="Secondary" Width="400" Height="400" Fill="Green"
                RelativePanel.Below="Primary" />
</RelativePanel>
```

3. Uncomment either of the setters.
4. Run the app on UWP and see that it works.
5. Run the app on WASM and see that after the splashscreen image disappears, you just get a blank screen and nothing showing the problem.

## Workaround

The simplest workaround is to not use the RelativePanel.
Still using a RelativePanel, it takes setting a lot of other properties to produce the same behavior.

## Environment

Nuget Package:

- [x] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [ ] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [ ] Uno.SourceGenerationTasks
- [ ] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other: 

Nuget Package Version(s): 3.7.4

Affected platform(s):

- [ ] iOS
- [ ] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

IDE:

- [ ] Visual Studio 2017 (version: )
- [x] Visual Studio 2019 (version: )
- [ ] Visual Studio for Mac (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )
- [ ] Visual Studio Code (version: )

Relevant plugins:

- [ ] Resharper (version: )

## Anything else we need to know?

Probably related to the issues also mentioned in #6120

**Comment by jeromelaban** [comment]
Thanks for the report. This will likely be fixed by #6120, as the current Uno implementation is close to the original, but definitely does not handle corner cases well.

Do you see the Rectangle content in the DOM tree in the debugging console (even if it does not appear in the page) ?

**Comment by mrlacey** [comment]
There's nothing in the page/DOM/source. :(
It took experimentation to find the cause of the problem.

**Comment by jeromelaban** [comment]
I see, the RelativePanel layout falls into a unstable state and never finishes measuring itself. This is definitely an issue that will be fixed by #6120. I'll include the sample in the other issue so we can track it.
